### ISSUE TITLE
ModuleNotFoundError: Module not found for Vue files

### ISSUE THREAD

**Issue by nmearl** [jupyterlab/jupyterlab#6254]
I am attempting to create a custom ipywidget jupyterlab extension that just mounts a small Vue js app on the front-end. Everything seems to work alright, I can get `yarn build` to run successfully, and I see an appropriately `index.js` file in the `dist` directory. When I look in the `lib` directory (defined as the `outDir` in the `tsconfig.json` file), I see both `ts` and `js` files for the jupyterlab entry points (e.g. `extension.ts`), as well as the ipywidgets file (`widget.ts`), and the compiled Vue single-file component files (e.g. `App.vue.d.ts`). 

However, the compiled `widget.js` file still references the `App.vue` file, which isn't generally an issue since in normal Vue applications, the `widget.js` and `App.vue` get bundled together in the `dist/index.js` file.

But when I run `jupyter labextension install .`, I always receive the error message

```
ModuleNotFoundError: Module not found: Error: Can't resolve './App.vue' in '/Users/nearl/anaconda3/envs/pits_dev/share/jupyter/lab/staging/node_modules/ipycubelab/lib'
```

I am using the [wigdet ts cookiecutter](https://github.com/jupyter-widgets/widget-ts-cookiecutter) template.

Any help or direction would be greatly appreciated.

**Comment by jasongrout** [comment]
Do you need to add the App.vue file to the package.json `files` attribute, to make sure it gets bundled with the npm package?

**Comment by jasongrout** [comment]
(at https://github.com/jupyter-widgets/widget-ts-cookiecutter/blob/0a0c36e021f2797c2d241abe9e23de8b3f00074c/%7B%7Bcookiecutter.github_project_name%7D%7D/package.json#L11-L14)

**Comment by jasongrout** [comment]
Oh, also note that the file trying to reference the App.vue file is in the `lib/` directory, not the `src/` directory. You'll need to copy the App.vue file to lib/ in the build step or prepublish step, or put the vue file somewhere where it can be referenced from the same relative path from both the .ts source in src/ and the .js compiled file in lib/

**Comment by nmearl** [comment]
@jasongrout Thank you so much for the quick response. I did in the past try to add the `src/**/*.vue` to the `files` attribute, but the `lib` directory still only contains the `App.vue.d.ts` compiled files, not the raw `.vue` file.

> Oh, also note that the file trying to reference the App.vue file is in the lib/ directory, not the src/ directory. 

Is there a way to tell `jupyter` to do an extra step, like copy the `App.vue` to the `lib` directory during install?

**Comment by nmearl** [comment]
> put the vue file somewhere where it can be referenced from the same relative path from both the .ts source in src/ and the .js compiled file in lib/

Ah! This is really clever. But adding `src/**/*.vue` to the `files` attribute, and changing the vue import to `import App from "../src/App.vue"` in the `widget.ts` file, I can successfully get past this error!

Thank you so much @jasongrout!

**Comment by nmearl** [comment]
However, now it doesn't seem that the jupyterlab build system can parse the `.vue` file:

```
ERROR in ./node_modules/ipycubelab/src/App.vue 1:0
Module parse failed: Unexpected token (1:0)
You may need an appropriate loader to handle this file type.
> <template>
|   <div id="app">
|     <img alt="Vue logo" src="./assets/logo.png">
 @ ./node_modules/ipycubelab/lib/widget.js 14:34-59
 @ ./node_modules/ipycubelab/lib/plugin.js
 @ ./build/index.out.js
 @ multi whatwg-fetch ./build/index.out.js
```

**Comment by jasongrout** [comment]
Right, webpack has no idea what to do with .Vue files. (I don't either...)

**Comment by nmearl** [comment]
Is this a webpack isssue? I include the necessary loaders for parsing and compiling vue component files -- and indeed, `yarn build` successfully produces the compiled javascript file, which can be mounted in a general html file and the vue app runs successfully.

But during the installation of the jupyter extension, it seems jupyter wants to the use the intermediate files compiled into the `lib` directory, not the end-result file in `dist` directory.

**Comment by blink1073** [comment]
We have a different set of loaders that isn't extensible because we were attempting to make the loading abstract so we wouldn't be tied to webpack long term.

**Comment by jasongrout** [comment]
Typically we suggest packages compile the file into plain javascript, css, etc, instead of relying on a webpack loader to do it for them in a jlab extension.

I don't know what .vue files do. Can you compile it into a javascript file, or something like that, in the lib directory?

**Comment by jasongrout** [comment]
You can point jupyterlab to use the dist/ directory. Right now JupyterLab looks into the lib directory because of this line: https://github.com/jupyter-widgets/widget-ts-cookiecutter/blob/0a0c36e021f2797c2d241abe9e23de8b3f00074c/%7B%7Bcookiecutter.github_project_name%7D%7D/package.json#L80

That line tells JupyterLab where the entry point for the jlab plugin is.
